examples/json.rs

```diff
@@ -46,86 +47,83 @@ fn parser() -> impl Parser<char, Json, Error = Simple<char>> {
                 .or(just('r').to('\r'))
                 .or(just('t').to('\t'))
                 .or(just('u').ignore_then(
-                    filter(|c: &char| c.is_digit(16))
+                    any()
+                        .filter(|c: &char| c.is_digit(16))
                         .repeated()
```

Comment:
I wonder whether we should make `text::digits` return a `Repeated<impl Parser>` instead of an `impl Parser` so we can do `text::digits(16).exactly(4)` here instead.

Reply:
Hm, that might not be a bad idea. Though it may want its own type, unless we make that change to `repeated` to default into slice collecting on `I::Token`

Reply:
Tbh, I think not allocating by default for text parsers is the way to go where possible. We should probably do the same for `whitespace`, `int`, etc. too